portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/BalloonEditor.testcase

```diff
@@ -0,0 +1,291 @@
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT",
+				paragraph_number = "1",
+				value1 = "text-align: justify;");
+		}
+	}
+
+	@description = "Assert toolbar with hyperlink icon is present"
+	@priority = "4"
+	test CanManageHyperlink {
+		task ("Click the hyperlinked text") {
+			Click(locator1 = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK");
+		}
+
+		task ("Assert the presence of the hyperlink toolbar elements") {
+			AssertElementPresent(locator1 = "CKEditor#HYPERLINK_UNLINK");
+
+			AssertElementPresent(locator1 = "CKEditor#HYPERLINK_TARGET");
+
+			AssertElementPresent(locator1 = "CKEditor#HYPERLINK_TEXT_INPUT");
+
+			AssertElementPresent(locator1 = "CKEditor#HYPERLINK_BROWSE_SRVR_BTN");
+
+			AssertElementPresent(locator1 = "CKEditor#CHK_BTN");
+		}
+
+		task ("Unlink") {
+			AssertElementPresent(locator1 = "CKEditor#HYPERLINK_UNLINK");
+
+			Click(locator1 = "CKEditor#HYPERLINK_UNLINK");
+
+			AssertElementNotPresent(
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK",
+				paragraph_number = "1");
+		}
+
+		task ("Link") {
+			TripleClick(
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT",
+				paragraph_number = "1");
+
+			Click(locator1 = "CKEditor#TOOLBAR_LINK_BUTTON_BALLOON");
+
+			Type.sendKeys(
+				locator1 = "CKEditor#HYPERLINK_TEXT_INPUT",
+				value1 = "www.liferay.com");
+
+			Click(locator1 = "CKEditor#CHK_BTN");
+
+			AssertVisible(
+				locator1 = "CKEditorBalloonEditorSample#BODY_TEXT_HYPERLINK",
+				paragraph_number = "1");
+		}
+	}
+
+	@description = "Assert table option enabled in the toolbar"
+	@priority = "4"
+	test CanManageTable {
+		task ("Create a table") {
+			BalloonEditor.clearParagraphWithBackspace(paragraph_number = "2");
+
+			BalloonEditor.addTable();
+
+			AssertVisible(locator1 = "CKEditorBalloonEditorSample#TABLE");
+		}
+
+		task ("Check Table Editor options") {
+			TripleClick(locator1 = "CKEditorBalloonEditorSample#TABLE_CELL");
+
+			AssertElementPresent(locator1 = "CKEditor#ROW_BUTTON");
+
+			AssertElementPresent(locator1 = "CKEditor#CELL_BUTTON");
+
+			AssertElementPresent(locator1 = "CKEditor#COLUMN_BUTTON");
+		}
+
+		task ("Write in a cell") {
+			var textSample = "test";
+
+			Click(locator1 = "CKEditorBalloonEditorSample#TABLE_CELL");
+
+			Type.sendKeys(
+				locator1 = "CKEditorBalloonEditorSample#TABLE_CELL",
+				value1 = "${textSample}");
+
+			AssertTextEquals(
+					locator1 = "CKEditorBalloonEditorSample#TABLE_CELL",
+					value1 = "${textSample}");
+		}
+
+		task ("Remove the table") {
+			TripleClick(locator1 = "CKEditorBalloonEditorSample#TABLE_CELL");
+
+			Click(locator1 = "CKEditor#REMOVE_TABLE_BUTTON");
+
+			AssertElementNotPresent(
+					locator1 = "CKEditorBalloonEditorSample#TABLE_CELL");
+		}
+	}
+
+	@description = "Assert text edition enabled in the table"
+	@priority = "4"
+	test CanEditTextInTable {
+		task ("Create a table") {
+			BalloonEditor.clearParagraphWithBackspace(paragraph_number = "2");
+
+			BalloonEditor.addTable();
```

Comment:
Tests are failing, please troubleshoot and fix:
- CanEditTextInTable
- CanManageHyperlink
- CanManageTable